xpra/platform/ui_thread_watcher.py

```diff
@@ -104,7 +107,9 @@ def UI_thread_wakeup(self, scheduled_at=0):
         self.last_UI_thread_time = monotonic_time()
         #UI thread was blocked?
         if self.UI_blocked:
-            log.info("UI thread is running again, resuming")
+            if self.announced_blocked:
+                log.info("UI thread is running again, resuming")
+                self.announced_blocked = False
```

Comment:
We suppress the `"UI thread is running again, resuming"` message here---`announced_blocked` will never be set unless `announce_timeout` has been exceeded.

Reply:
Sorry, my message was not very clear at all!

If `announced_blocked` is not set, then no message is printed - is this intended?

Maybe I misunderstand the symptoms you were seeing?
Can you perhaps post a more complete log sample in #3035 so I can see how often the messages show up?

Reply:
Hey, yeah, I wasn't sure I was understanding your message.  Here's some log output:

    2021-03-02 07:03:14,377 client   2 @03.245 Xpra GTK3 X11 server version 3.0.13-r28292 64-bit
    2021-03-02 07:03:14,377 client   2 @03.245  running on Linux Debian testing bullseye
    2021-03-02 07:03:14,391 client   2 @03.258 Attached to socket:///run/user/1001/xpra/REDACTED-67
    2021-03-02 07:03:14,391 client   2 @03.258  (press Control-C to detach)
    2021-03-02 07:03:14,877 client   2 @03.745 server does not support xi input devices
    2021-03-02 07:03:14,877 client   2 @03.746  server uses: xtest
    2021-03-02 07:03:15,176 New unix-domain connection received
    2021-03-02 07:03:15,176  on '/run/user/1001/xpra/REDACTED-67'
    2021-03-02 07:03:15,177 New unix-domain connection received
    2021-03-02 07:03:15,177  on '/home/REDACTED/.xpra/REDACTED-67'
    2021-03-02 07:03:15,383 client   2 @04.252 Error: cannot access the list of printers
    2021-03-02 07:03:15,384 client   2 @04.252  failed to connect to server
    2021-03-02 07:08:44,496 client   2 @33.364 UI thread is now blocked
    2021-03-02 07:08:48,269 client   2 @37.137 UI thread is running again, resuming
    2021-03-02 07:08:55,999 client   2 @44.868 UI thread is now blocked
    2021-03-02 07:08:56,696 client   2 @45.565 UI thread is running again, resuming

To answer your question directly:

> If `announced_blocked` is not set, then no message is printed - is this intended?

Yes.  I'd like to suppress both the block and resume notifications, since---as I understand it---they are part of the normal operation of XPRA on my system when DPMS engages (at least on my machine).

The block and resume messages always appear in pairs (unbalanced messages should be impossible, right?).

I chose this approach (adding a new environment variable, and not changing the default behavior) because it sounds like these notifications serve a very useful diagnostic role for many people.  If these messages were less important for diagnostics, we might consider changing the loglevel from `.info` to `.debug`, and scrapping this patch.